src/Models/Laravel/Wallet.php

```diff
@@ -98,4 +98,11 @@ protected static function boot()
 
         self::observe(new WalletObserver());
     }
+
+    protected function pivotIdentifier(): Attribute
+    {
+        return Attribute::make(
+            get: fn () => $this->id,
```

Comment:
This should be fine as long as the wallets table doesn't get truncated and rebuilt, otherwise the IDs will likely change as they do for other on-chain data.

Reply:
I don't think this get's truncated. Also we don't have any other options since public_key and external_id is nullable